### PR TITLE
TST/CLN: Fixturize tests/frame/test_quantile.py

### LINKED ISSUE #22471: TST/CLN: remove TestData from frame-tests; replace with fixtures
Following review in #22236: 
> ok, pls open a new issue that refs this, to remove use of `TestData` in favor of fixtures

Started the process in that PR by creating a `conftest.py` that translates all the current attributes of `TestData` to fixtures. Need to incrementally replace their usages in `pandas/tests/frame/`.

### PR DESCRIPTION
**Pull request by makbigc**
-  xref #22471
-  passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/tests/frame/test_quantile.py [hunk 1]
```diff
@@ -3,24 +3,23 @@
...
+class TestDataFrameQuantile:
 
-    def test_quantile(self):
+    def test_quantile(self, datetime_frame):
         from numpy import percentile
 
-        q = self.tsframe.quantile(0.1, axis=0)
-        assert q['A'] == percentile(self.tsframe['A'], 10)
-        tm.assert_index_equal(q.index, self.tsframe.columns)
+        q = datetime_frame.quantile(0.1, axis=0)
```
Comment (simonjayhawkins) [inline pandas/tests/frame/test_quantile.py]: a pattern that is now being used across the codebase is to 'rename' the fixture at the beginning of the test (it facilitates the use of more descriptive fixture names)

 so just add `df = datetime_frame` at the beginning of the test and then use `df` throughout the test.

File: pandas/tests/frame/test_quantile.py [hunk 2]
```diff
@@ -97,19 +96,19 @@ def test_quantile_axis_parameter(self):
         with pytest.raises(ValueError, match=msg):
             df.quantile(0.1, axis="column")
 
-    def test_quantile_interpolation(self):
+    def test_quantile_interpolation(self, datetime_frame, int_frame):
```
Comment (simonjayhawkins) [inline pandas/tests/frame/test_quantile.py]: rather that having 2 fixtures that are tested independently, would it be straightforward to split this test?

**Review by simonjayhawkins — COMMENTED** [review]
@makbigc Thanks for the PR. a couple of suggestions.

File: pandas/tests/frame/test_quantile.py [hunk 3]
```diff
@@ -155,6 +143,27 @@ def test_quantile_interpolation(self):
...
+    def test_quantile_interpolation_int(self, int_frame):
+        # see gh-10174
+
+        df = int_frame
+        # interpolation = linear (default case)
+        q = df.quantile(0.1)
+        assert q['A'] == np.percentile(df['A'], 10)
+
+        # test with and without interpolation keyword
+        q1 = df.quantile(0.1)
```
Comment (simonjayhawkins) [inline pandas/tests/frame/test_quantile.py]: what's the difference between q and q1?

**Comment by simonjayhawkins** [comment]
@makbigc does this close #22471?

**Comment by makbigc** [comment]
Yes, close one of the issues there

**Comment by simonjayhawkins** [comment]
> Yes, close one of the issues there

OK, have changed to xref #22471.

File: pandas/tests/frame/test_quantile.py (hunk 3 above)
Comment (makbigc) [reply pandas/tests/frame/test_quantile.py]: They are the same. q1 removed.

Comment (simonjayhawkins) [reply pandas/tests/frame/test_quantile.py]: probably it's best to leave it as is at the moment. could add a TODO though.

File: pandas/tests/frame/test_quantile.py [hunk 4]
```diff
@@ -155,6 +143,28 @@ def test_quantile_interpolation(self):
...
+
+    def test_quantile_interpolation_int(self, int_frame):
+        # see gh-10174
+
+        df = int_frame
+        # interpolation = linear (default case)
+        q = df.quantile(0.1)
+        assert q['A'] == np.percentile(df['A'], 10)
+
+        # TODO: Remove the following, as q1 is not different from q
```
Comment (simonjayhawkins) [inline pandas/tests/frame/test_quantile.py]: sorry to be pedantic here, can you change this to 

`# TODO: q1 is not different from q`

and move it below 

`# test with and without interpolation keyword`

**Review by simonjayhawkins — CHANGES_REQUESTED** [review]
@makbigc lgtm. just the comment regarding the TODO.

**Review by simonjayhawkins — APPROVED** [review]
@jreback

**Comment by jreback** [comment]
thanks @makbigc

**Pull request merged**